### DISCUSSION TITLE
[ClientWebSocket] When I received all messages from websocket server

### DISCUSSION THREAD

**Discussion by BethanyZhou** [dotnet/runtime#59738] [Q&A]
Hi All,

I'm receiving message from websocket server. I have no idea when I received all messages from server. In my casae, websocket will not close after sending message and it will send 3 message to client, each message is marked EndOfMessage as true. Any flag I can detect all messages are received? So I can send a message to server?
```c#
                var allBytes = new List<byte>();
                WebSocketReceiveResult result;
                while (socket.State == WebSocketState.Open)
                {

                    var rcvBytes = new byte[BUFF_SIZE];
                    var rcvBuffer = new ArraySegment<byte>(rcvBytes);
                    result = socket.ReceiveAsync(rcvBuffer, CancellationToken.None).GetAwaiter().GetResult();
                    allBytes.AddRange(rcvBuffer.Slice(0, result.Count));
                };
                Console.WriteLine($"Recieved: {Encoding.UTF8.GetString(allBytes.ToArray(), 0, allBytes.Count)}");
```

Could you kindly help to answer help to implement this functionality?
Really appreciate if this discussion is able to get a quick response.

**Comment by davidfowl** [comment]
What does all messages mean? The result returned from the ReceiveAsync has the type of message that was received (text, binary or close). You should be able to use the close frame to detect if the server has gracefully closed the connection.

**Reply by BethanyZhou** [reply]
Thank you for your response @davidfowl. In my case, ,server will respond one more messages to client, each message is marked EndOfMessage as true.  websocket will not close after sending message.
I have no idea about the number of messages. Is there any flag I can detect all messages are received so that I can send a message to server? Actually, it's an interactive communication connection.
